### PR TITLE
Fixed bug in VectorTools::get_position_vector.

### PR DESCRIPTION
**Pull request by luca-heltai**
This should fix issue #1138. The solution won't work for all combination of FiniteElements, but it should be fairly general for all *primitive* ones. 

The others, at the moment, won't work with MappingFEField either, so this should not be an issue right now....

### CODE REVIEW

File: include/deal.II/numerics/vector_tools.templates.h [hunk 1]
```diff
@@ -7177,33 +7177,91 @@ namespace VectorTools
...
+        // solve the following system:
+        //
+        // v_j phi_j(q_i) = w_k psi_k(q_i) = w_k delta_ki = w_i
+        //
+        // where psi_k are the basis functions for fe_q, and phi_i are
+        // the basis functions of the target space.
+        //
+        // Morally, we should invert the matrix T_ij = phi_i(q_j),
+        // however in general this matrix is not invertible, since
+        // there may by components which do not contribute to the
```
Comment (tjhei) [inline include/deal.II/numerics/vector_tools.templates.h]: by- > be

File: include/deal.II/numerics/vector_tools.templates.h [hunk 2]
```diff
@@ -7177,33 +7177,91 @@ namespace VectorTools
...
+        // If index is not the same as feq.dofs_per_cell, we won't
+        // know how to invert the resulting matrix. Bail out.
+        Assert(index == feq.dofs_per_cell, ExcNotImplemented());
+
+        for (unsigned int j=0; j<fe.dofs_per_cell; ++j)
           {
-            unsigned int comp_i = fe.system_to_component_index(i).first;
-            if (fe_mask[comp_i])
-              for (unsigned int j=0; j<points.size(); ++j)
+            unsigned int comp_j = fe.system_to_component_index(j).first;
```
Comment (tjhei) [inline include/deal.II/numerics/vector_tools.templates.h]: const

**Comment by tjhei** [comment]
Looks good! :+1: 

- If it doesn't work for non-primitive elements, can you check this with an ```Assert(.., ExcInternalError("This function is only implemented for primitive elements."));``` please?
- Can you update the documentation of the function to state when it (doesn't) work?
- Changelog?

**Comment by luca-heltai** [comment]
@tjhei : done.

**Pull request merged**